zfsbootmenu/lib/zfsbootmenu-lib.sh

```diff
@@ -404,7 +404,10 @@ snapshot_dispatcher() {
 
     [ -n "${user_input}" ] || return
 
-    valid_name=$( echo "${user_input}" | tr -c -d 'a-zA-Z0-9-_.:' )
+    shopt -s extglob
+    valid_name="${user_input//+([!a-zA-Z0-9-_.:])/}"
```

Comment:
The set complement operator in bash is `!` and not `^`?!?

Reply:
`^` anchors the pattern to the start of the string. The best documentation I've found for extglob is at https://mywiki.wooledge.org/glob#extglob .

Reply:
Outside a character class, sure. But in character classes, usually a `^` immediately after the opening bracket means the class should be inverted.

Reply:
https://wiki.bash-hackers.org/syntax/pattern#bracket_expressions

`^` and `!` are equivalent in this context. I can change it if you'd like.

Reply:
Not at all; I was just surprised.